helix-core/src/main/java/org/apache/helix/controller/stages/MessageGenerationPhase.java

```diff
@@ -127,7 +127,9 @@ public void process(ClusterEvent event) throws Exception {
             Message message =
                 createMessage(manager, resourceName, partition.getPartitionName(), instanceName,
                     currentState, nextState, sessionIdMap.get(instanceName), stateModelDef.getId(),
-                    resource.getStateModelFactoryname(), bucketSize);
```

Comment:
can we simply pass in the entire resource and createMessage can fetch required attributes from resource